include/pms-utils/vdb/vdb_parsers.hpp

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include "pms-utils/atom/atom.hpp"
+#include "pms-utils/atom/atom_parser.hpp"
+#include "pms-utils/ebuild/ebuild.hpp"
+#include "pms-utils/ebuild/ebuild_parser.hpp"
+#include "pms-utils/misc/x3_utils.hpp"
+#include "pms-utils/vdb/vdb.hpp"
+
+#include <boost/parser/parser.hpp>
+#include <cstdint>
+#include <filesystem>
+#include <optional>
+#include <vector>
+
+namespace pms_utils::parsers::vdb {
+
+namespace _internal {
+
+// many of these parsers need to cope with empty files and trailing newlines that may or
+// may not exist?
+
+PARSER_RULE_T(md5, std::string);
+PARSER_DEFINE(md5, boost::parser::repeat(32)[boost::parser::hex_digit]);
+
+PARSER_RULE_T(size, std::uint64_t);
+PARSER_DEFINE(size, boost::parser::ulong_long >> *boost::parser::ws);
```

Comment:
I don't like these being lenient with whitespaces *and* being used in other parsers, that's prone to error